### PR TITLE
New Module: BCF switch

### PR DESCRIPTION
**Pull request by tedelhourani**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

bcf_switch

##### ANSIBLE VERSION

```
ansible 2.3.0 (bcf 69f17bb39c) last updated 2017/03/13 19:08:53 (GMT -700)
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by jayakody** [comment]
ACK

**Comment by jayakody** [comment]
LGTM. 
shipit

**Comment by vuile** [comment]
shipit

**Comment by jctanner** [comment]
bot_status

File: lib/ansible/modules/network/bigswitch/bcf_switch.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: bcf_switch
+author: "Ted (@tedelhourani)"
+short_description: Create and remove a bcf switch.
+description:
+    - Create and remove a bcf switch.
+version_added: "2.3"
```
Comment (gundalow) [inline lib/ansible/modules/network/bigswitch/bcf_switch.py]: `version_added: "2.4"`

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by gundalow** [comment]
Closing and reopening to rerun CI

File: lib/ansible/modules/network/bigswitch/bigmon_chain.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # -*- coding: utf-8 -*-
```
Comment (gundalow) [inline lib/ansible/modules/network/bigswitch/bigmon_chain.py:2]: Modules should have permission 644 (not 775), could you please check this.

File: lib/ansible/modules/network/bigswitch/bcf_switch.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: bcf_switch
+author: "Ted (@tedelhourani)"
+short_description: Create and remove a bcf switch.
```
Comment (gundalow) [inline lib/ansible/modules/network/bigswitch/bcf_switch.py:30]: One bit of feed back we've recently had fun the community is to put the vendor and/or product name in the `short_description` so a more searchable string will appear in http://docs.ansible.com/ansible/list_of_network_modules.html

So I think bigmon needs adding to this.

File: lib/ansible/modules/network/bigswitch/bcf_switch.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            name=dict(type='str', required=True),
+            fabric_role=dict(choices=['spine', 'leaf'], required=True),
+            leaf_group=dict(type='str', required=False),
+            mac=dict(type='str', required=True),
+            controller=dict(type='str', required=True),
+            state=dict(choices=['present', 'absent'], default='present'),
+            validate_certs=dict(type='bool', default='True'),
+            access_token=dict(type='str', no_log=True)
+        )
+    )
```
Comment (gundalow) [inline lib/ansible/modules/network/bigswitch/bcf_switch.py:176]: What would it take for the module to support "check mode", ie be idempotent?

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/bigswitch/bcf_switch.py (hunk 3 above)
Comment (tedelhourani) [reply lib/ansible/modules/network/bigswitch/bcf_switch.py:30]: This is a bcf (big cloud fabric), not a big mon module.

File: lib/ansible/modules/network/bigswitch/bcf_switch.py (hunk 4 above)
Comment (tedelhourani) [reply lib/ansible/modules/network/bigswitch/bcf_switch.py:176]: Can be implemented now as the module is already idempotent.

But before, what does this mean?

lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.changed.: expected a dictionary
lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.invocation.description: required key not provided @ data['description']. Got None
lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.invocation.module_args: extra keys not allowed @ data['module_args']. Got {'validate_certs': False, 'fabric_role': 'leaf', 'controller': '192.168.86.230', 'mac': '00:00:00:02:00:02', 'name': 'Rack1Leaf1', 'state': 'present', 'access_token': None, 'leaf_group': 'R1'}
lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.invocation.returned: required key not provided @ data['returned']. Got None
lib/ansible/modules/network/bigswitch/bcf_switch.py:0:0: E319 RETURN.invocation.type: required key not provided @ data['type']. Got None
click here for bot help

**Comment by tedelhourani** [comment]
Closing this one starting from scratch.

**Pull request closed without merging**